### ISSUE TITLE
Fix S2183: False positive for 8-bit and 16-bit integers

### ISSUE THREAD

**Issue by valhristov** [SonarSource/sonar-dotnet#2016]
### Description

[RSPEC-2183](https://jira.sonarsource.com/browse/RSPEC-2183)

When shifting 8-bit and 16-bit integers the rules for 32-bit apply, which means that the shift count operand is first truncated to its lowest 5 bits and then the operation is performed.

```
short x = some_number;
short result_01 = (short)(x >> 1);
short result_17 = (short)(x >> 17); // False positive
short result_33 = (short)(x >> 33); // Noncompliant
```

Note: RSPEC is already up to date.

Original report and analysis here: 
https://community.sonarsource.com/t/csharpsquid-s2183-bit-shifting-false-positive-and-wrong-explanation/2995
